### PR TITLE
[VirtRegMap] Replace a single value enum with a static constexpr member variable. NFC

### PR DESCRIPTION
**Pull request by topperc**
Change the constant to INT_MAX instead of our own large number. Any value larger than a valid frame index should work.

I'm a bit puzzled why it was using a shift of 30. A long time ago when it was first created, the value was INT_MAX. Then it was changed in e2b77d57c0c13 to (~0 >> 1) which I guess was trying to be INT_MAX without using the constant. But ~0 is an `int` so that produced -1.

I'm not sure what the 'l' suffix was for. Unless that was an attempt to avoid undefined behavior had the shift been 31 instead of 30. But 'long' is 32 bits on some targets so that wouldn't have worked for all platforms.

Using INT_MAX is straightforward.

### CODE REVIEW

File: llvm/include/llvm/CodeGen/VirtRegMap.h [hunk 1]
```diff
@@ -32,9 +32,7 @@ class TargetInstrInfo;
 
   class VirtRegMap : public MachineFunctionPass {
   public:
-    enum {
-      NO_STACK_SLOT = (1L << 30)-1,
-    };
+    static constexpr unsigned NO_STACK_SLOT = (1u << 30) - 1;
```
Comment (MatzeB) [inline llvm/include/llvm/CodeGen/VirtRegMap.h:35]: judging by the return type of `getStackSlot` this should be `int` instead?

**Review by MatzeB — APPROVED** [review]
LGTM with changed type.

File: llvm/include/llvm/CodeGen/VirtRegMap.h (hunk 1 above)
Comment (MatzeB) [reply llvm/include/llvm/CodeGen/VirtRegMap.h:35]: While on it maybe remove this `public:` section here and merge it with the one below?

Comment (topperc) [reply llvm/include/llvm/CodeGen/VirtRegMap.h]: > judging by the return type of `getStackSlot` this should be `int` instead?

Yep. I had unsigned on the brain, then changed it but forgot to commit it.

**Comment by topperc** [comment]
@MatzeB do you think it's possible to use INT_MAX instead of our own number here?

**Comment by qcolombet** [comment]
> @MatzeB do you think it's possible to use INT_MAX instead of our own number here?

I don't remember if this value needs to match what we have for describe registers, but if so we need to stick to it.
See `llvm/include/llvm/CodeGen/Register.h`
```
  //   [2^30;2^31) Stack slots. (Rarely used.)
```

@topperc double check if it needs to be connected and if not `INT_MAX` may be fine.

**Comment by topperc** [comment]
> > @MatzeB do you think it's possible to use INT_MAX instead of our own number here?
> 
> I don't remember if this value needs to match what we have to describe the virtual registers, but if so we need to stick to it. See `llvm/include/llvm/CodeGen/Register.h`
> 
> ```
>   //   [2^30;2^31) Stack slots. (Rarely used.)
> ```
> 
> @topperc double check if it needs to be connected and if not `INT_MAX` may be fine.

It looks like VirtRegMap stores the incoming stack slots numbers directly without converting them to this Register form. And I don't see anything that converts them to that form. So I think this constant just needs to be something that will never be a valid stack slot number.

**Comment by MatzeB** [comment]
At a first glance it seems like users in `Register` should always call `Register::index2StackSlot` / `Register::stackSlot2Index` and if someone were to pass `NO_STACK_SLOT` to it, it would overflow anyway.

`INT_MAX` seems fine to me as there's no test failures...

**Pull request merged**